src/main/java/com/google/api/codegen/util/SymbolTable.java

```diff
@@ -89,4 +114,24 @@ private String getAndSaveSuffix(String desiredName) {
     symbolTable.add(desiredName + Integer.toString(i));
     return Integer.toString(i);
   }
+
+  /**
+   * Returns the next underscore suffix that makes desiredName unique.
+   *
+   * <p>Stores the joined desiredName/suffix in an internal map. Assumes that desiredName is already
+   * in the symbolTable.
+   *
+   * <p>For example, if "foo" is passed, "_" is returned. If "foo" is passed again, "__" is
```

Comment:
Good point, I'll adjust it...

Incidentally, do you think it's useful to support underscores at all? Having written this PR, I'm not sure if underscores are necessarily more idiomatic in Ruby/Node.js

Is there a better strategy besides numbers/underscores you can think of off the top of your head?